uefi/src/proto/shell/mod.rs

```diff
@@ -54,4 +55,82 @@ impl Shell {
         let dir_ptr: *const Char16 = directory.map_or(ptr::null(), |x| x.as_ptr());
         unsafe { (self.0.set_cur_dir)(fs_ptr.cast(), dir_ptr.cast()) }.to_result()
     }
+
+    /// Gets the environment variable or list of environment variables
+    ///
+    /// # Arguments
+    ///
+    /// * `name` - The environment variable name of which to retrieve the
+    ///   value
+    ///   If None, will return all defined shell environment
+    ///   variables
+    ///
+    /// # Returns
+    ///
+    /// * `Some(Vec<env_value>)` - Value of the environment variable
+    /// * `Some(Vec<env_names>)` - Vector of environment variable names
+    /// * `None` - Environment variable doesn't exist
+    #[must_use]
+    pub fn get_env<'a>(&'a self, name: Option<&CStr16>) -> Option<Vec<&'a CStr16>> {
```

Comment:
I thought you changed that? [Context.](https://github.com/rust-osdev/uefi-rs/pull/1679/commits/6547011b76e5c58a25233dc00f75f133a7cc398c#r2129055664)

At this point, I expected to have
- `var(name: &CStr16>) -> Option<&CStr16> {}`
- `vars() -> Vars` (Iterator over all vars)
- `set_var(name: &CStr16>, value: &CStr16) -> Option<&CStr16> {}`

All three functions should work akin to [this](https://doc.rust-lang.org/stable/std/env/fn.var.html), [this](https://doc.rust-lang.org/stable/std/env/fn.vars.html), and [this](https://doc.rust-lang.org/stable/std/env/fn.set_var.html).